sentry-android-core/src/main/java/io/sentry/android/core/AndroidUserCache.java

```diff
@@ -0,0 +1,90 @@
+package io.sentry.android.core;
+
+import android.content.SharedPreferences;
+import io.sentry.core.IUserCache;
+import io.sentry.core.protocol.User;
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+// TODO: to think, this could be a SentryOptionsCache that could be expanded, does it make sense?
+
+/** the Android User Cache class. It caches the user data in the Android SharedPreferences */
+public final class AndroidUserCache implements IUserCache {
+
+  // TODO: only these 3 fields make sense to me, otherwise caching a raw json would be better, but I
+  // don't want to do that.
+  private static final String EMAIL = "email";
+  private static final String ID = "id";
+  private static final String USERNAME = "user_name";
+
+  private final @NotNull SharedPreferences sharedPreferences;
+  private final @NotNull SentryAndroidOptions options;
+  private final @Nullable String sentryDeviceId;
+
+  public AndroidUserCache(
+      final @NotNull SentryAndroidOptions options,
+      final @NotNull SharedPreferences sharedPreferences,
+      final @Nullable String sentryDeviceId) {
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required.");
+    this.sharedPreferences =
+        Objects.requireNonNull(sharedPreferences, "SharedPreferences is required.");
+    this.sentryDeviceId = sentryDeviceId;
+  }
+
+  //  TODO: I believe this makes sense only for global hub SDKs, but if not,
+  //   how do we cache multiple users (because we have multiple scopes)?
+
+  @Override
+  public void setUser(final @Nullable User user) {
+    if (options.isCacheUserForSessions()) {
+      final SharedPreferences.Editor edit = sharedPreferences.edit();
+      if (user != null) {
+        edit.putString(ID, user.getId())
```

Comment:
Could we settle for only caching the `userId` instead? Or at least a single value (when a user is set, we evaluate the which value based on an order (id>username>email)? Would be worth seeing what the issues page already considers when accounting for "user" of a crash.

If we're expecting to store the actual User, we'd need to consider the protocol can change (versioning/new fields added), [user define data](https://docs.sentry.io/development/sdk-dev/event-payloads/user/#attributes) and makes it trickier:

> All other keys are stored as extra information but not specifically processed by Sentry.

If we're storing just a single value to be used for session tracking only, it would minimize these edge cases.
Everything else you said stands true:

`setUser(...)` makes the SDK evaluate a new value for session `distrinctId`. `setUser(null)` deletes that cached value.

Reply:
yep, a single value would be ok, makes sense.
the order would be `id` > `email` > `username` if following the release health specs.